test/metrics/prom-metrics-linter/custom_linter_rules.go

```diff
@@ -43,14 +43,6 @@ func CustomLinterRules(problems []promlint.Problem, mf *dto.MetricFamily, operat
 		})
 	}
 
-	// Check "_timestamp_seconds" suffix for non-counter metrics
-	if *mf.Type != dto.MetricType_COUNTER && strings.HasSuffix(*mf.Name, "_timestamp_seconds") {
-		problems = append(problems, promlint.Problem{
-			Metric: *mf.Name,
-			Text:   "non-counter metric should not have \"_timestamp_seconds\" suffix",
-		})
-	}
-
```

Comment:
@avlitman why was this rule added? Maybe we have some reference that suggested the need for this rule, no?

Reply:
this was asked by Shirly in the initial ticket

Reply:
In kubevirt the _timestamp_seconds it's only counters: https://github.com/kubevirt/kubevirt/blob/021fbac594c7d093d8ab8fb318ecd1667028e0d7/docs/metrics.md?plain=1#L96

Reply:
Actually @avlitman @assafad , I asked this because the prom-linter linter failed for the `timestamp_seconds` when we used counter, if my memory is correct. I don't mind allowing it to be gauge, but since the timestamp always increases then it make more sense from my POV to have it as counter.
WDYT?

Reply:
I'll open a bug for the other metrics.
/lgtm

Reply:
but remove this test or change it to gauge?

Reply:
I think that for now let's only remove it, because we have a few counters with timestamps, so it will block Kubevirt CI. Later, if needed we can add it. 
We can also try first to push it to Prometheus docs and enforce it in [promlint](https://github.com/prometheus/client_golang/tree/main/prometheus/testutil/promlint), since I don't think it's related to Kubevirt/CNV specifically.

Reply:
>Actually @avlitman @assafad , I asked this because the prom-linter failed for the timestamp_seconds when we used counter, if my memory is correct.

@sradco Yes, it's still relevant. This is why I'm not removing this part: https://github.com/kubevirt/monitoring/blob/main/test/metrics/prom-metrics-linter/custom_linter_rules.go#L54-L65

Once we change all the timestamp metrics to gauges, this part can be removed as well, and we can add a restriction to only allow gauges use this suffix.

Reply:
I prefer to add gauge as an option in this test, then once metric names change to delete the counter.
But I don't have strong opinion about it, just make more sense to me. 
and the custom_linter_rules.go is for custom rules we want to add, it can be anything basically, but I agree with trying to ask them.

Reply:
>I prefer to add gauge as an option in this test, then once metric names change to delete the counter.

Then, you only restrict histogram/summary metrics to not use this suffix. I don't see why it's specific to our needs (like the `kubevirt_suboperator` prefix is). If Prometheus maintainers tell us that all types should use this suffix, why do we need to enforce something? Do we have any special usecase?

Reply:
sure so let's ask them and after they change it or say it's not needed- because if it is needed we don't want new metric will be added with wrong suffix in the meanwhile- that's my concern.

Reply:
@sradco mention that we need to change counter with these suffix. no?